main/test/flix/Test.Exp.Fixpoint.Stratification.flix

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2025 Casper Dalgaard Nielsen
+ *                Adam Yasser Tallouzi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+mod Test.Exp.Fixpoint.Stratification {
+
+    @Test
+    def testStratification01(): Bool = {
+        let r = #{
+            A1(1).
+            B1(x) :- A1(x).
+            C1(x) :- B1(x).
+            D1(x) :- C1(x).
+        };
+        let solved = solve r;
+        Vector#{1} `Assert.eq` query solved select (x) from A1(x) and
+            Vector#{1} `Assert.eq` query solved select (x) from B1(x) and
+            Vector#{1} `Assert.eq` query solved select (x) from C1(x) and
+            Vector#{1} `Assert.eq` query solved select (x) from D1(x)
```

Comment:
Maybe unindent then we can see that the lines are the same except for A1, B1 and so on.

Reply:
We can make a follow-up PR where we reformat the tests